modules/custom/entity_access_by_field/src/EntityAccessHelper.php

```diff
@@ -133,6 +133,7 @@ public static function entityAccessCheck(
             return self::NEUTRAL;
           }
           elseif (
+            !$entity->isNew() &&
```

Comment:
@volodymyr-sydor Could you explain more why do we need this check?
How we can make sure that it doesn't cause any regression for cases with both new and not new entities?

Reply:
Currently we have: "Unexpected error when creating event assigned to a specific group"

So, during the creation a new entity there is called the current `entityAccessCheck` where we have:

`$group_contents = GroupContent::loadByEntity($entity);` 

but a new entity doesn't have an ID yet which caused that error, that's why we should prevent such cases

Also, this check we have for `getEntityAccessResult()` but for `entityAccessCheck()` method is missed:

    public static function getEntityAccessResult(
      EntityInterface $entity,
      string $operation,
      AccountInterface $account,
      string $permission = NULL
    ): AccessResultInterface {
      $access = self::entityAccessCheck(

     ....

    // If the social_event_invite module is enabled and a person got invited
    // then allow access to view the node.
    // @todo Come up with a better solution for this code.
    if (
      \Drupal::moduleHandler()->moduleExists('social_event_invite') &&
      $entity->getEntityTypeId() === 'node' &&
      !$entity->isNew() &&
      $operation === 'view'
    ) {